core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/StorageLocation.java

```diff
@@ -261,4 +261,44 @@ public ChangeLog createChangeLog(String summary, String columnsChanged, User use
     return change;
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((alias == null) ? 0 : alias.hashCode());
+    result = prime * result + ((creationTime == null) ? 0 : creationTime.hashCode());
+    result = prime * result + ((creator == null) ? 0 : creator.hashCode());
+    result = prime * result + (int) (id ^ (id >>> 32));
+    result = prime * result + ((identificationBarcode == null) ? 0 : identificationBarcode.hashCode());
+    result = prime * result + ((locationUnit == null) ? 0 : locationUnit.hashCode());
+    result = prime * result + ((parentLocation == null) ? 0 : parentLocation.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)